runner/utils/driver/istio/names.go

```diff
@@ -31,6 +32,29 @@ func getInternalServiceDomain(projectID, serviceID, version string) string {
 	return fmt.Sprintf("%s.%s.svc.cluster.local", getInternalServiceName(serviceID, version), projectID)
 }
 
+func checkIfInternalServiceDomain(projectID, serviceID, internalServiceDomain string) bool {
+	arr := strings.Split(internalServiceDomain, ".")
+	if strings.HasSuffix(internalServiceDomain, "svc.cluster.local") {
+		if len(arr) == 5 && arr[1] == projectID && strings.HasPrefix(arr[0], serviceID) {
```

Comment:
Dont do prefix match for service id. Split by `-` and do exact match for service.